plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/SizeBasedSplitWeightProvider.java

```diff
@@ -23,30 +23,24 @@
 public class SizeBasedSplitWeightProvider
         implements HiveSplitWeightProvider
 {
-    private static final SplitWeight STANDARD_WEIGHT = SplitWeight.standard();
-
-    private final SplitWeight minimumWeight;
+    private final double minimumWeight;
     private final double sizeToWeightRatio;
 
-    public SizeBasedSplitWeightProvider(SplitWeight minimumWeight, DataSize targetSplitSize)
+    public SizeBasedSplitWeightProvider(double minimumWeight, DataSize targetSplitSize)
     {
-        this.minimumWeight = requireNonNull(minimumWeight, "minimumWeight is null");
-        long minimumWeightValue = minimumWeight.getValue();
-        checkArgument(minimumWeightValue > 0 && minimumWeightValue <= STANDARD_WEIGHT.getValue(), "minimumWeight must be > 0 and <= %s, found: %s", STANDARD_WEIGHT, minimumWeight);
-        double targetSizeInBytes = (double) requireNonNull(targetSplitSize, "targetSplitSize is null").toBytes();
-        this.sizeToWeightRatio = 1.0 / (targetSizeInBytes / STANDARD_WEIGHT.getValue());
+        checkArgument(Double.isFinite(minimumWeight) && minimumWeight > 0 && minimumWeight <= 1, "minimumWeight must be > 0 and <= 1, found: %s", minimumWeight);
+        this.minimumWeight = minimumWeight;
+        long targetSizeInBytes = requireNonNull(targetSplitSize, "targetSplitSize is null").toBytes();
+        checkArgument(targetSizeInBytes > 0, "targetSplitSize must be > 0, found: %s", targetSplitSize);
+        this.sizeToWeightRatio = 1 / (double) targetSizeInBytes;
+        checkArgument(Double.isFinite(sizeToWeightRatio) && sizeToWeightRatio > 0, "invalid target split size to weight ratio: %s", sizeToWeightRatio);
     }
 
     @Override
     public SplitWeight weightForSplitSizeInBytes(long splitSizeInBytes)
     {
-        long computedWeight = (long) Math.ceil(splitSizeInBytes * sizeToWeightRatio);
-        if (computedWeight <= minimumWeight.getValue()) {
-            return minimumWeight;
-        }
-        if (computedWeight >= STANDARD_WEIGHT.getValue()) {
-            return STANDARD_WEIGHT;
-        }
-        return SplitWeight.of(computedWeight);
+        double computedWeight = splitSizeInBytes * sizeToWeightRatio;
```

Comment:
Why not treat weight as just the "size in bytes" (or "size in kilobytes" or another unit if we want to avoid potential overflows when adding weights -- although that seems unlikely), instead trying to normalize to a targetSize dynamically. It would simplify the code a bit.